pkgs/gnomod/gnomod.go

```diff
@@ -0,0 +1,210 @@
+package gnomod
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/gnolang/gno/pkgs/gnolang"
+	"github.com/gnolang/gno/pkgs/std"
+	"golang.org/x/mod/modfile"
+	"golang.org/x/mod/module"
+)
+
+var (
+	source        = "test3.gno.land:36657" // "127.0.0.1:26657"
+	queryPathFile = "vm/qfile"
+)
+
+// ReadModFile reads, parses and validates the mod file at gnomod.
+func ReadModFile(absModPath string) (f *File, err error) {
+	data, err := os.ReadFile(absModPath)
+	if err != nil {
+		return nil, fmt.Errorf("reading gno.mod: %s", err)
+	}
+
+	f, err = Parse(absModPath, data)
+	if err != nil {
+		return nil, fmt.Errorf("parsing gno.mod: %s", err)
+	}
+
+	return f, err
+}
+
+func IsModFileExist(absModPath string) bool {
+	_, err := os.Stat(absModPath)
+	return err == nil
+}
+
+// GetGnoModPath returns the path for gno modules
+func GetGnoModPath() (string, error) {
+	goPath := os.Getenv("GOPATH")
```

Comment:
`go env GOMOD` is more complex than just looking at `$GOPATH/pkg/mod`, it will dynamically check for `$GOMOD`, and fallback to build `$GOPATH/pkg/mod`, and even have a sane default when there is no env var.

I suggest you look at an official helper making the same than `go env GOMOD`.

Reply:
-> https://github.com/golang/go/blob/932d0ae83ea8ce9adb2b23b28788b860447b1f61/src/cmd/doc/dirs.go#L192

Can be done in another PR.